src/Compilers/CSharp/Test/Emit/Emit/LocalStateTracing/LocalStateTracingTests.cs

```diff
@@ -0,0 +1,5898 @@
+    IL_0069:  nop
+    // sequence point: Microsoft.CodeAnalysis.Runtime.Instrumentation.FlushPayload();
+    IL_006a:  ldloc.2
+    IL_006b:  ldc.i4.2
+    IL_006c:  ldc.i4.1
+    IL_006d:  stelem.i1
+    IL_006e:  call       ""void Microsoft.CodeAnalysis.Runtime.Instrumentation.FlushPayload()""
+    IL_0073:  nop
+    // sequence point: }
+    IL_0074:  leave.s    IL_007f
+  }
+  finally
+  {
+    // sequence point: <hidden>
+    IL_0076:  ldloca.s   V_0
+    IL_0078:  call       ""void Microsoft.CodeAnalysis.Runtime.LocalStoreTracker.LogReturn()""
+    IL_007d:  nop
+    IL_007e:  endfinally
+  }
+  // sequence point: }
+  IL_007f:  ret
+}
+");
+        }
+
+        [Fact]
+        public void Composition_LocalStateTracing_TestCoverage_LocallySuppressed()
+        {
+            var source = WithHelpers(@"
+class C
+{
+    static void Main()
+    {
+        [System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverage]
+        void F()
+        {
+            int a = 1;    
+        }
+
+        F();
+
+        Microsoft.CodeAnalysis.Runtime.Instrumentation.FlushPayload();
+    }
+}
+") + CSharpInstrumentationChecker.InstrumentationHelperSource;
+
+            var verifier = CompileAndVerify(
+                source,
+                options: TestOptions.UnsafeDebugExe,
+                emitOptions: EmitOptions.Default
+                    .WithInstrumentationKinds(ImmutableArray.Create(InstrumentationKind.LocalStateTracing, InstrumentationKind.TestCoverage)),
+                verify: s_verification,
+                targetFramework: s_targetFramework);
+
+            verifier.VerifyMethodBody("C.<Main>g__F|0_0", @"
+{
+  // Code size       41 (0x29)
+  .maxstack  3
+  .locals init (Microsoft.CodeAnalysis.Runtime.LocalStoreTracker V_0,
+                int V_1) //a
+  // sequence point: <hidden>
+  IL_0000:  ldtoken    ""void C.Main()""
+  IL_0005:  ldtoken    ""void C.<Main>g__F|0_0()""
+  IL_000a:  call       ""Microsoft.CodeAnalysis.Runtime.LocalStoreTracker Microsoft.CodeAnalysis.Runtime.LocalStoreTracker.LogLambdaEntry(int, int)""
+  IL_000f:  stloc.0
+  .try
+  {
+    // sequence point: {
+    IL_0010:  nop
+    // sequence point: int a = 1;
+    IL_0011:  ldloca.s   V_0
+    IL_0013:  ldc.i4.1
+    IL_0014:  dup
+    IL_0015:  stloc.1
+    IL_0016:  ldc.i4.1
+    IL_0017:  call       ""void Microsoft.CodeAnalysis.Runtime.LocalStoreTracker.LogLocalStore(uint, int)""
+    IL_001c:  nop
+    // sequence point: }
+    IL_001d:  leave.s    IL_0028
+  }
+  finally
+  {
+    // sequence point: <hidden>
+    IL_001f:  ldloca.s   V_0
+    IL_0021:  call       ""void Microsoft.CodeAnalysis.Runtime.LocalStoreTracker.LogReturn()""
+    IL_0026:  nop
+    IL_0027:  endfinally
+  }
+  // sequence point: }
+  IL_0028:  ret
+}
+");
+        }
+
+        [Fact]
+        public void EmitDifference()
+        {
+            var source = WithHelpers(@"
+class C
+{
+    static void F()
```

Comment:
In most tests here, it looks like we could change `static C.F()` to `static C.Main()` and verify expected output.